src/persist/benches/consensus.rs

```diff
@@ -0,0 +1,73 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+//! Benchmarks for different persistent Write implementations.
+
+use std::path::Path;
+use std::time::{Duration, Instant};
+
+use criterion::measurement::WallTime;
+use criterion::{Bencher, BenchmarkGroup, BenchmarkId, Throughput};
+
+use mz_persist::mem::MemConsensus;
+use mz_persist::sqlite::SqliteConsensus;
+use mz_persist::storage::{Consensus, SeqNo, VersionedData};
+use mz_persist::workload::{self, DataGenerator};
+
+fn new_sqlite_consensus(shard: &str, parent: &Path) -> SqliteConsensus {
+    let consensus_path = parent.join(shard);
+    SqliteConsensus::open(&consensus_path, shard.to_string())
+        .expect("creating a SqliteConsensus cannot fail")
+}
+
+// Benchmark the write throughput of Consensus::compare_and_set.
+fn bench_compare_and_set<C: Consensus>(consensus: &mut C, data: Vec<u8>, b: &mut Bencher) {
+    let deadline = Instant::now() + Duration::from_secs(3600);
+    let mut prev = futures_executor::block_on(consensus.head(deadline))
```

Comment:
lets set an initial value here instead, in case that ends up being different performance-wise in some impl of Consensus. concretely, something like the following

```
// Set an initial value in case that has different performance characteristics
let mut current = futures_executor::block_on(consensus.compare_and_set(deadline, None, VersionedData{seqno: SeqNo(1), data: data.clone})).expect("failed to set initial value")
```

optional: I think `current` reads a tiny bit better as a name for this than `prev`?

Reply:
done and done.